src/components/admin/PosterConfiguration/PosterConfiguration.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { useCallback, useMemo } from "react";
+import { useForm, useFormState } from "react-hook-form";
+import { useHistory } from "react-router-dom";
+import { useAsyncFn } from "react-use";
+import { faPen } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Section } from "components/admin/Section";
+import { SectionHeading } from "components/admin/SectionHeading";
+import { SectionTitle } from "components/admin/SectionTitle";
+import { parse } from "csv-parse/browser/esm/sync";
+import * as Yup from "yup";
+
+import { ADMIN_IA_SPACE_EDIT_PARAM_URL } from "settings";
+
+import { PosterDetails, uploadPosters } from "api/admin";
+
+import { SpaceWithId, WorldId } from "types/id";
+
+import { generateUrl } from "utils/url";
+
+import { useManagedSpaces } from "hooks/spaces/useManagedSpaces";
+import { useWorldParams } from "hooks/worlds/useWorldParams";
+
+import { Button } from "../Button";
+import { Input } from "../Input";
+import { InputGroup } from "../InputGroup";
+import { TablePanel } from "../TablePanel";
+
+const csvUploadSchema = Yup.object().shape({
+  csvFile: Yup.mixed().required("File is required"),
+});
+
+interface ChannelConfigurationProps {
+  space: SpaceWithId;
+}
+
+export const PosterConfiguration: React.FC<ChannelConfigurationProps> = ({
+  space,
+}) => {
+  const { worldSlug } = useWorldParams();
+  const history = useHistory();
+  const { register, handleSubmit, control } = useForm({
+    reValidateMode: "onChange",
+    resolver: yupResolver(csvUploadSchema),
+  });
+  const { errors } = useFormState({ control });
+  const { managedSpaces, isLoading: isLoadingSpaces } = useManagedSpaces({
+    worldId: space.worldId as WorldId,
+    spaceId: space.id,
+  });
+
+  const [
+    { loading: isUpdating, error: updateError, value: result },
+    updatePosters,
+  ] = useAsyncFn(async (data) => {
+    const csvFiles = data.csvFile;
+    if (!csvFiles.length) {
+      throw new Error("Must provide a file to upload");
+    }
+    const fileContents = await new Promise<string>((resolve, reject) => {
```

Comment:
I think we'd benefit if this parsing logic was separated from the actual render
For now, can we please add `@debt` comments, so we can clean it up once we have time for it?